app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace App\Http\Controllers;
+
+
+use App\Models\User;
+use App\Models\Authentication;
+use App\Http\Controllers\Controller;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Mail;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Foundation\Auth\RegistersUsers;
+use Illuminate\Auth\Events\Registered;
+use Illuminate\Support\Facades\DB;
+use Cookie;
+use App\Http\Requests\UserRequest;
+
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    //urlを押した後に開く登録用の画面
+    public function index(Request $request)
+    {
+        $token = $request->token;  
+        $authentication = Authentication::where('token', $token)->first();
+
+        $datetime1 = now();      //現在時間 Carbon('now'); を省略した形
+        $datetime2 = $authentication->expired_at;  
+
+       if ($datetime1 < $datetime2 ) {
+       } elseif ($datetime1 > $datetime2) {
+            echo '期限切れです'; 
+       }
+
+       Cookie::queue('token', $token);
+
+        return view('register');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(UserRequest $request)
+    {
+        $token = $request->cookie('token'); 
+        $authentication = Authentication::where('token',$token)->first(); 
+        $email = $authentication->email;  
+
+        $user = new User;
+        $user->email = $authentication->email;
+        $user->nickname = $request->nickname;
+        $user->first_name = $request->first_name;
+        $user->last_name = $request->last_name;
+        $user->first_name_kana = $request->first_name_kana;
+        $user->last_name_kana = $request->last_name_kana;
+        $user->gender = $request->gender;
+        $user->postal_code = $request->postal_code;
+        $user->prefecture = $request->prefecture;
+        $user->city = $request->city;
+        $user->house_number = $request->house_number;
+        $user->building_name = $request->building_name;
+        $user->birthday = $request->birthday;
+        $user->telephone_number = $request->telephone_number;
+        $user->password = Hash::make($request->Password);
```

Comment:
```suggestion
        $user->password = Hash::make($request->password);
```
場所によっては大文字小文字が違うだけでエラーが起きますので、注意してください。

Reply:
指摘箇所を修正しました。
今後注意していきます。